errors/structured_error.go

```diff
@@ -15,50 +15,61 @@ func (e StructuredError) Error() string {
 	return e.Message
 }
 
-func NewStructuredError(errorDetails []types.ErrorDetails) *StructuredError {
+func NewStructuredError(errorDetails []types.ErrorDetails) error {
 	// "{severity}: {name} ({code}) - {source}, {description}, resolution: {resolution} at {location} see {helpLink}"
 	message := strings.Builder{}
-	for _, error := range errorDetails {
+	for _, e := range errorDetails {
 		if message.Len() > 0 {
 			message.WriteString("\n")
 		}
-		formatErrorDetails(&message, error)
+		if err := formatErrorDetails(&message, e); err != nil {
+			return fmt.Errorf("failed to format error details: %w", err)
+		}
```

Comment:
I don't think we should raise the error if we fail to parse the error message. For the users it's not useful. We should probably just return the original error message unparsed.